exasol_integration_test_docker_environment/lib/docker/images/create/utils/file_directory_list_hasher.py

```diff
@@ -129,22 +188,18 @@ def collect_files_and_directories(
                     f"Directory: {directory} contains symlink loops (Symlinks pointing to a parent directory). Please fix!")
             inodes.add(stat.st_ino)
 
-            if self.hash_directory_names:
-                new_directories = [d for d in dirs if not self.is_excluded_directory(d)]
-                tmp_collected_directories.extend([(directory, os.path.join(root, d)) for d in new_directories])
-                numCharacters += sum([len(d) for d in new_directories])
+            new_directories = [os.path.join(root, d) for d in dirs if not self.is_excluded_directory(d)]
+            directory_handler(new_directories)
+            numCharacters += sum([len(d) for d in new_directories])
```

Comment:
Hm, that is a bit weird. Did we really wanted to sum up the characters of all path we traversed. Does this make really sense. Do you remember our reasoning. If we keep it as is, we at least need to add a comment.

Reply:
And would it be more plausible to count the number of visited dirs and files instead.

Reply:
I think, one reason was, we didn't want to assume what a valid depth for a path is

Reply:
I also don't remember the reason. 
The ticket (#97) also does not provide more info.

Reply:
I had a look into the PR, it looks like, we tried a few things, and I think this discussion caused it
https://github.com/exasol/integration-test-docker-environment/pull/98#discussion_r674528536
However, the current implementation doesn't make sense, anyway, because the error is only for one path

Reply:
I think, we can extract this check then also to the Symlink checker or so.